include/fiction/io/write_sqd_sim_result.hpp

```diff
@@ -0,0 +1,301 @@
+     *
+     * @param value The value to convert.
+     * @return The string representation of the value.
+     */
+    [[nodiscard]] std::string any_to_string(const std::any& value) const noexcept
+    {
+        static const std::unordered_map<std::type_index, std::function<std::string(const std::any&)>> converters{
+            {typeid(int8_t), [](const std::any& v) { return std::to_string(std::any_cast<int8_t>(v)); }},
+            {typeid(uint8_t), [](const std::any& v) { return std::to_string(std::any_cast<uint8_t>(v)); }},
+            {typeid(int16_t), [](const std::any& v) { return std::to_string(std::any_cast<int16_t>(v)); }},
+            {typeid(uint16_t), [](const std::any& v) { return std::to_string(std::any_cast<uint16_t>(v)); }},
+            {typeid(int32_t), [](const std::any& v) { return std::to_string(std::any_cast<int32_t>(v)); }},
+            {typeid(uint32_t), [](const std::any& v) { return std::to_string(std::any_cast<uint32_t>(v)); }},
+            {typeid(int64_t), [](const std::any& v) { return std::to_string(std::any_cast<int64_t>(v)); }},
+            {typeid(uint64_t), [](const std::any& v) { return std::to_string(std::any_cast<uint64_t>(v)); }},
+            {typeid(float), [](const std::any& v) { return std::to_string(std::any_cast<float>(v)); }},
+            {typeid(double), [](const std::any& v) { return std::to_string(std::any_cast<double>(v)); }},
+            {typeid(long double), [](const std::any& v) { return std::to_string(std::any_cast<long double>(v)); }},
+            {typeid(std::string), [](const std::any& v) { return std::any_cast<std::string>(v); }},
+            {typeid(const char*), [](const std::any& v) { return std::string(std::any_cast<const char*>(v)); }},
+            {typeid(char), [](const std::any& v) { return std::string(1, std::any_cast<char>(v)); }},
+        };
+
+        const auto& type = value.type();
+        const auto& it   = converters.find(type);
+
+        return it == converters.end() ? std::string() : it->second(value);
+    }
+    /**
+     * Converts a charge distribution to a string.
+     *
+     * The string is a concatenation of the charge states of the cells in the distribution in order.
+     *
+     * The charge states are represented as follows:
+     * - NEUTRAL: '-'
+     * - NEGATIVE: '0'
+     * - POSITIVE: '1'
+     *
+     * @param surface The charge distribution to represent as a string.
+     * @return The string representation of the charge distribution.
+     */
+    std::string charge_distribution_to_string(const charge_distribution_surface<Lyt>& surface) const
+    {
+        std::stringstream stream{};
+
+        static const auto charge_to_string = [](const sidb_charge_state& state) -> char
+        {
+            switch (state)
+            {
+                case sidb_charge_state::NEUTRAL: return '-';
+                case sidb_charge_state::NEGATIVE: return '0';
```

Comment:
Are you really sure with this?

Reply:
Nope, I'm not :laughing:

Reply:
I am pretty sure that it has to be the expected convention.